src/python/pants/backend/terraform/goals/deploy_test.py

```diff
@@ -3,89 +3,15 @@
-from pants.backend.terraform.dependencies import TerraformInitRequest, TerraformInitResponse
 from pants.backend.terraform.goals.deploy import DeployTerraformFieldSet
-from pants.backend.terraform.goals.deploy import rules as terraform_deploy_rules
-from pants.backend.terraform.target_types import TerraformDeploymentTarget, TerraformModuleTarget
-from pants.core.goals import deploy
+from pants.backend.terraform.testutil import rule_runner_with_auto_approve, standard_deployment
 from pants.core.goals.deploy import Deploy, DeployProcess
-from pants.core.register import rules as core_rules
-from pants.core.util_rules import source_files
-from pants.engine import process
 from pants.engine.internals.native_engine import Address
-from pants.engine.rules import QueryRule
 from pants.testutil.rule_runner import RuleRunner, mock_console
 
-
-@pytest.fixture
-def rule_runner() -> RuleRunner:
-    rule_runner = RuleRunner(
-        target_types=[TerraformModuleTarget, TerraformDeploymentTarget],
-        rules=[
-            *dependency_inference.rules(),
-            *dependencies.rules(),
-            *tool.rules(),
-            *terraform_deploy_rules(),
-            *source_files.rules(),
-            *deploy.rules(),
-            *core_rules(),
-            *process.rules(),
-            QueryRule(DeployProcess, (DeployTerraformFieldSet,)),
-            QueryRule(TerraformInitResponse, (TerraformInitRequest,)),
-        ],
-        preserve_tmpdirs=True,
-    )
-    rule_runner.set_options(["--download-terraform-args='-auto-approve'"])
-    return rule_runner
-
-
-@dataclass
-class StandardDeployment:
-    files: dict[str, str]
-    state_file: Path
-    target: Address = Address("src/tf", target_name="stg")
-
-
-@pytest.fixture
-def standard_deployment(tmpdir) -> StandardDeployment:
-    # We have to forward "--auto-approve" to TF because `mock_console` is noninteractive
-    state_file = Path(str(tmpdir.mkdir(".terraform").join("state.json")))
-    return StandardDeployment(
-        {
-            "src/tf/BUILD": textwrap.dedent(
-                """\
-                terraform_deployment(
-                    name="stg",
-                    var_files=["stg.tfvars"],
-                    backend_config="stg.tfbackend",
-                    root_module=":mod",
-                )
-                terraform_module(name="mod")
-            """
-            ),
-            "src/tf/main.tf": textwrap.dedent(
-                """\
-                terraform {
-                    backend "local" {
-                        path = "/tmp/will/not/exist"
-                    }
-                }
-                variable "var0" {}
-                resource "null_resource" "dep" {}
-                """
-            ),
-            "src/tf/stg.tfvars": "var0=0",
-            "src/tf/stg.tfbackend": f'path="{state_file}"',
-        },
-        state_file,
-    )
+rule_runner = rule_runner_with_auto_approve
```

Comment:
If you import a fixture, the linter goes mad because it's being addressed in a test (used among arguments). To avoid adding a pragma, I assign rule_runner to be that fixture to keep the test arguments consistent across the codebase.

Reply:
This won't be necessary for any other test violation fix - this is the only place this needs to happen.